### ISSUE TITLE
[Chapter13] error: error getting orders:  {result: "failed", error: "businessNetwork: Card not found: "}

### ISSUE THREAD

**Issue by diegoduncan21** [rddill-IBM/ZeroToBlockchain#28]
I deployed, tested and completed the demo on ibm kubernetes without problems, but after a while an error appears: `error getting orders:  {result: "failed", error: "businessNetwork: Card not found: "}` I thougth it was because of the free cluster, but after deploying on paid cluster, the error appeared again. Any suuggestion about how to solve this error?

**Comment by rddill-IBM** [comment]
Not yet. I've recently noticed the same phenomena on my cluster and need to look at the logs to determine why. I'll respond here when I have more insight.

**Comment by diegoduncan21** [comment]
Thank you Bob.

**Comment by rddill-IBM** [comment]
It appears that the ~/.hfc-key-store folder goes missing after some period of time, or some system event, from the cloud foundry part of this app. I'll spend some time on this over the next day or so and respond here.

**Comment by diegoduncan21** [comment]
Thanks Bob! Looking forward to hearing from you soon.

**Comment by Limcooo** [comment]
Hi both, I am running the chapter13 on Mac and when I run ./install-bx-cli.sh I am getting following stuff. Any chance you have encounter something like this? 
```

./install-bx-cli.sh: line 29: showstep: command not found
updating for OSX
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 51.3M  100 51.3M    0     0  5607k      0  0:00:09  0:00:09 --:--:-- 6635k
       =====================================================
       -----> making kubectl executable
       =====================================================
       =====================================================
       -----> moving kubectl to correct location
       =====================================================
       =====================================================
       -----> installing cloud Foundry CLI
       =====================================================
./install-bx-cli.sh: line 8: wget: command not found
sudo: apt-key: command not found
tee: /etc/apt/sources.list.d/cloudfoundry-cli.list: No such file or directory
deb https://packages.cloudfoundry.org/debian stable main
sudo: apt-get: command not found
sudo: apt-get: command not found
       =====================================================
       -----> Installing BM Cloud (Bluemix) CLI
       =====================================================
Unrecognized Linux platform: darwin. Quit installation.
       =====================================================
       -----> Getting list of uninstalled IBM Cloud CLI Packages
       =====================================================
       =====================================================
       -----> Installing packages
       =====================================================
Installing 
FAILED
Incorrect Usage.

NAME:
   install - Install CLI plug-in

USAGE:
   bx plugin install PLUGIN_NAME [-r REPO_NAME] [-v VERSION] [-f]
   bx plugin install LOCAL-PATH/TO/PLUGIN | URL [-f]

If no repository is specified, the command uses the official plug-in repository 'Bluemix'.
If no version is specified, the command selects the latest available version to install.

EXAMPLE:
   bx plugin install http://example.com/downloads/my-plugin
   bx plugin install ~/Downloads/my-plugin
   bx plugin install plugin-echo
   bx plugin install plugin-echo -v 1.0.0
   bx plugin install plugin-echo -r Bluemix -v 1.0.0

OPTIONS:
   -r value  Repo name where the plug-in binary is located
   -v value  Version of the plug-in to be installed
   -f        Force install of plugin without confirmation

```

When I tried to run line 29 in the file it seems it is working. But when I tried to run line 8 
`wget -q -O - https://packages.cloudfoundry.org/debian/cli.cloudfoundry.org.key | sudo apt-key add -
`

it's giving me back command not found. Thank you in advance!

**Comment by diegoduncan21** [comment]
hi @Limcooo, i think that is better to do a new issue for your problem, because is a different than mine.

**Comment by diegoduncan21** [comment]
gently reminder

**Comment by ChiragETRM** [comment]
Hi Bob! I'm facing the same issue, please help!

**Comment by ChiragETRM** [comment]
z2b-buyer.js:132 {result: "failed", error: " order ericbornonthecloudinccom1531862807322 add f…ection Card not found: eric@bornonthecloudinc.com"}

**Comment by ChiragETRM** [comment]
Hi Bob, 

Thanks for this. Testing this script and will get back to you shortly! Highly appreciate your help!

Chirag

**Comment by rddill-IBM** [comment]
I am rebuilding this on a paid cluster and will test shortly. My free cluster expired, so I have to test on a paid cluster. I have found some errors in the scripts I'm using from the ibm-container-service repo. I don't believe that they affect the issue we've all encountered. 

The root cause seems to be that the wallet is disappearing and the wallet is, I believe, in the cloud foundry space not in the cluster. If so, then when the cloud foundry app restarts, that wallet goes away. I'm guessing right now because of my own inability to test. For those of you who can test, the wallet should be in ```.hfc-key-store```. If that folder on your cloud foundry system is empty except for the network and business network ids, then the cloud foundry app was reset. This clearing and reset action is performed by the init.sh script which is run each time the cloud foundry app restarts. 

If that is the root cause, then we can remedy it quickly by replacing the init.sh script at the end of it's execution with a new script which does not perform the delete and copy functions.

**Comment by ChiragETRM** [comment]
Hi Bob,

I retried copying the above script and running through the steps to install the network on Bluemix/Kubernetes. However, i get the following error at the end of the kubernetes-deploy.sh command:

       =====================================================
       -----> Please enter your root password for this O/S now
       =====================================================
cp: cannot stat 'controller/restapi/features/composer/creds/-priv': No such file or directory

**Comment by rddill-IBM** [comment]
that's correct. once the script has completed it's work with the peerAdmin.card. it deletes it to avoid having confusing data in your system.

**Comment by rddill-IBM** [comment]
This appears to be the result of the nodejs app being restarted in the cloud. When this happens, the local card store is emptied. Working on a fix for it, will close this issue when fix is posted.
